tests/testsctp.nim

```diff
@@ -0,0 +1,102 @@
+# Nim-WebRTC
+# Copyright (c) 2024 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+{.used.}
+
+import chronos
+import ../webrtc/udp_transport
+import ../webrtc/stun/stun_transport
+import ../webrtc/dtls/dtls_transport
+import ../webrtc/sctp/sctp_transport
+import ../webrtc/sctp/sctp_connection
+import ./asyncunit
+
+suite "SCTP":
+  teardown:
+    checkLeaks()
+
+  type
+    SctpStackForTest = object
+      localAddress: TransportAddress
+      udp: UdpTransport
+      stun: Stun
+      dtls: Dtls
+      sctp: Sctp
+
+  proc initSctpStack(localAddress: TransportAddress): SctpStackForTest =
+    result.localAddress = localAddress
+    result.udp = UdpTransport.new(result.localAddress)
+    result.stun = Stun.new(result.udp)
+    result.dtls = Dtls.new(result.stun)
+    result.sctp = Sctp.new(result.dtls)
+    result.sctp.listen()
+
+  proc closeSctpStack(self: SctpStackForTest) {.async: (raises: [CancelledError]).} =
+    await self.sctp.stop()
+    await self.dtls.stop()
+    await self.stun.stop()
+    await self.udp.close()
+
+  asyncTest "Two SCTP nodes connecting to each other, then sending/receiving data":
+    var
+      sctpServer = initSctpStack(initTAddress("127.0.0.1:4444"))
```

Comment:
It's better to use 0 port to avoid errors with port reuse.

Reply:
I changed those. It makes me change some things in `udp_transport.nim` and Stun transport/connection. If you think it's not appropriate to make those changes in this PR, I can revert these changes and create another one.